src/tests/ftest/util/yaml_utils.py

```diff
@@ -0,0 +1,469 @@
+"""
+(C) Copyright 2020-2023 Intel Corporation.
+
+SPDX-License-Identifier: BSD-2-Clause-Patent
+"""
+from collections import OrderedDict
+import os
+import re
+import yaml
+
+from ClusterShell.NodeSet import NodeSet
+
+
+class YamlException(BaseException):
+    """Base exception for this module."""
+
+
+def get_test_category(test_file):
+    """Get a category for the specified test using its path and name.
+
+    Args:
+        test_file (str): the test python file
+
+    Returns:
+        str: concatenation of the test path and base filename joined by dashes
+
+    """
+    file_parts = os.path.split(test_file)
+    return "-".join([os.path.splitext(os.path.basename(part))[0] for part in file_parts])
+
+
+def get_yaml_data(yaml_file):
+    """Get the contents of a yaml file as a dictionary.
+
+    Removes any mux tags and ignores any other tags present.
+
+    Args:
+        yaml_file (str): yaml file to read
+
+    Raises:
+        YamlException: if an error is encountered reading the yaml file
+
+    Returns:
+        dict: the contents of the yaml file
+
+    """
+    class DaosLoader(yaml.SafeLoader):  # pylint: disable=too-many-ancestors
+        """Helper class for parsing avocado yaml files."""
+
+        def forward_mux(self, node):
+            """Pass on mux tags unedited."""
+            return self.construct_mapping(node)
+
+        def ignore_unknown(self, node):  # pylint: disable=no-self-use,unused-argument
+            """Drop any other tag."""
+            return None
+
+    DaosLoader.add_constructor('!mux', DaosLoader.forward_mux)
+    DaosLoader.add_constructor(None, DaosLoader.ignore_unknown)
+
+    yaml_data = {}
+    if os.path.isfile(yaml_file):
+        with open(yaml_file, "r", encoding="utf-8") as open_file:
+            try:
+                yaml_data = yaml.load(open_file.read(), Loader=DaosLoader)
+            except yaml.YAMLError as error:
+                raise YamlException(f"Error reading {yaml_file}") from error
+    return yaml_data
+
+
+def find_values(obj, keys, key=None, val_type=str):
+    """Find dictionary values of a certain type specified with certain keys.
+
+    Args:
+        obj (obj): a python object; initially the dictionary to search
+        keys (list): list of keys to find their matching list values
+        key (str, optional): key to check for a match. Defaults to None.
+
```

Comment:
You might want to add description for `val_type`

Reply:
Done, thanks.